spec/controllers/comments_controller_spec.rb

```diff
@@ -0,0 +1,247 @@
+require 'spec_helper'
+
+describe CommentsController do
+  include LoginMacros
+  include RedirectExpectationHelper
+
+  let(:comment) { create(:comment) }
+  let(:unreviewed_comment) { create(:unreviewed_comment) }
+
+  before(:each) do
+    request.env["HTTP_REFERER"] = "/where_i_came_from"
+  end
+
+  describe "GET #add_comment_reply" do
+    context "when comment is unreviewed" do
+      it "redirects logged out user to login path with an error" do
+        get :add_comment_reply, comment_id: unreviewed_comment.id
+        it_redirects_to_with_error(login_path, "Sorry, you cannot reply to an unapproved comment.")
+      end
+
+      it "redirects logged in user to root path with an error" do
+        fake_login
+        get :add_comment_reply, comment_id: unreviewed_comment.id
+        it_redirects_to_with_error(root_path, "Sorry, you cannot reply to an unapproved comment.")
+      end
+    end
+
+    context "when comment is not unreviewed" do
+      it "redirects to the comment on the commentable without an error" do
+        get :add_comment_reply, comment_id: comment.id
+        expect(flash[:error]).to be_nil
+        expect(response).to redirect_to(work_path(comment.ultimate_parent, show_comments: true, anchor: "comment_#{comment.id}"))
+      end
+
+      it "redirects to the comment on the commentable with the reply form open and without an error" do
+        get :add_comment_reply, comment_id: comment.id, id: comment.id
+        expect(flash[:error]).to be_nil
+        expect(response).to redirect_to(work_path(comment.ultimate_parent, add_comment_reply_id: comment.id, show_comments: true, anchor: "comment_#{comment.id}"))
+      end
+    end
+  end
+
+  describe "GET #unreviewed" do
```

Comment:
Could we add a test for this action where the `comment_id` param doesn't correspond to an actual comment, so that it hits the `if @commentable.nil?` block on line 165 in comments_controller?